### DISCUSSION TITLE
PrimeNG- p-table Issue with virtual scroll

### DISCUSSION THREAD

**Discussion by 861576** [primefaces/community#1141] [PrimeNG]
We have implemented p-table and as our data is large we implemented virtual scroll with rowsize 20, As we keep scrolling the data loads without any issues but when we are at row 50 and want to remove the 50th row, it is not working, The table reloads and scrolls up to row 1.  should that not retain the previous position

Also the row removed from the array is still displayed in UI
This issue is only when [virtualScroll]="true" when false its working as expected

PFB the snippet used to delete the row,

this.displayRows= this.displayRows.filter(row=>(row && row .dataid &&  row .dataid !=='700157550578' ));

Kindly guide us on the gaps.

Issue found in higher versions of primeng v16.3.1 and above
It was working at v11.4.4

**Comment by cetincakiroglu** [comment]
Hi,

In v11.4.4 we were using angular cdk and we replaced it with our custom scroller component in v14. Could you please share a stackblitz example so we can clearly identify the issue?

**Reply by 861576** [reply]
https://stackblitz.com/edit/oqn5zi-qo55zx?file=src%2Fapp%2Fdemo%2Ftable-virtual-scroll-lazy-demo.ts

**Reply by 861576** [reply]
Hope the above stackblitz link  helps,

Steps to recreate.
1. On load of table , scroll down to 50th row
2. Click on clear button
3. Table reloads and takes to 1st row.
4. scroll down to see the removed row still visible in  row 50

One other observation
1. On load of table, click the clear button of 3rd row
2. scroll down
3. table doesnot load after a while

**Reply by 861576** [reply]
@cetincakiroglu any update on the above 
Guidance/work around would be much appreciated as its impacting our application

**Reply by cetincakiroglu** [reply]
Hi,

Sorry for the late response, I've reviewed your example and realized that loadCarsLazy method was invoked after you removed the car by hitting the clear button which re-created the virtualCars from the cars array. You also need to remove item from the cars array too. More complex logic is up to you. Here is a stackblitz example https://stackblitz.com/edit/oqn5zi-yn5mcf?file=src%2Fapp%2Fdemo%2Ftable-virtual-scroll-lazy-demo.ts

**Reply by 861576** [reply]
@cetincakiroglu Even in the example you have shared ( https://stackblitz.com/edit/oqn5zi-yn5mcf?file=src%2Fapp%2Fdemo%2Ftable-virtual-scroll-lazy-demo.ts ) I'm able to recreate the issue

Steps to recreate.

1. On load of table , scroll down to 50th row
2. Click on clear button
3. Table reloads and takes to 1st row.

should it not stay there in the same position


One other observation

1. Relaunch the app 
2. On load of table, click the clear button of 3rd row
3. scroll down
4. table doesnot load after a while(loads only the first 40 records)

**Reply by 861576** [reply]
@cetincakiroglu any update plz....
